### ISSUE TITLE
Serializing IInspectable to RNW

### ISSUE THREAD

**Issue by asklar** [microsoft/react-native-winrt#39]
## Context
I'm working on react-native-xaml which allows projecting XAML types for use in RNW apps via the ViewManager API, and allows apps to use these XAML types directly in their JSX markup.

One of the features that is important to support is setting up event handlers from JS; these event handlers need to be able to get event args.
Currently the support in react-native-xaml for event args is fairly limited. Basically the way this works is we try to serialize what we consider important properties on every relevant type. For example, for every UIElement we will serialize the Name and Tag properties, for every object we will serialize the runtime class name, for MenuFlyoutItem we would serialize the Text properties, for SelectionChangedEventArgs we serialize what things were added and removed to the selection, etc.

## Feature request
I'd like to eventually get out of the business of having to do this serialization manually, and instead rely on jswinrt for it. 
Basically I'd like to have a native API from jswinrt that looks like:
```cpp
void JSWinRT::Serialize(
    const winrt::Microsoft::ReactNative::JSValueWriter& writer, 
    const winrt::Windows::Foundation::IInspectable& obj)
```
